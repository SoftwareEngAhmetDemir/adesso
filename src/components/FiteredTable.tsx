import { store } from '../redux/store';
import UnitsTables from './UnitsTables';

function FiteredTable() {
  let { status } = store.getState();
  return (
    <div>
      <div className="tab-content" id="pills-tabContent">
        <div
          className={`tab-pane fade ${status === 'ALL' ? 'show active' : ''} `}
          id="pills-home"
          role="tabpanel"
          aria-labelledby="pills-home-tab"
          tabIndex={0}
        >
          <UnitsTables />
        </div>
        <div
          className={`tab-pane fade ${status === 'Dark' ? 'show active' : ''} `}
          id="pills-profile"
          role="tabpanel"
          aria-labelledby="pills-profile-tab"
          tabIndex={0}
        >
          <UnitsTables />
        </div>
        <div
          className={`tab-pane fade ${
            status === 'Feudal' ? 'show active' : ''
          } `}
          id="pills-contact"
          role="tabpanel"
          aria-labelledby="pills-contact-tab"
          tabIndex={0}
        >
          <UnitsTables />
        </div>
        <div
          className={`tab-pane fade ${
            status === 'Castle' ? 'show active' : ''
          } `}
          id="pills-disabled"
          role="tabpanel"
          aria-labelledby="pills-disabled-tab"
          tabIndex={0}
        >
          <UnitsTables />
        </div>
        <div
          className={`tab-pane fade ${
            status === 'Imperial' ? 'show active' : ''
          } `}
          id="pills-disabled2"
          role="tabpanel"
          aria-labelledby="pills-disabled-tab2"
          tabIndex={0}
        >
          <UnitsTables />
        </div>
      </div>
    </div>
  );
}

export default FiteredTable;
