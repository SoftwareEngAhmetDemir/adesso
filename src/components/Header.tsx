import { NavLink } from 'react-router-dom';
interface HeaderTypes {
  title: string;
}
function Header({ title }: HeaderTypes) {
  return (
    <header id="header">
      <nav className="navbar navbar-expand-lg  bg-white d-flex justify-content-between pt-3">
        <div className="container-fluid custom-navbar">
          <h1 className="title">{title}</h1>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <div className="d-flex" role="search">
              <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                <li className="nav-item">
                  <NavLink className="nav-link" to="/">
                    Home
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink className="nav-link" to="/units">
                    Units
                  </NavLink>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </nav>
    </header>
  );
}

export default Header;
