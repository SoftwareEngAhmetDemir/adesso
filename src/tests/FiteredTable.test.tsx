import { shallow } from 'enzyme';
import Header from '../components/Header';
import FiteredTable from '../components/FiteredTable';

test('should test FilteredTable component', () => {
 const wrapper = shallow(<FiteredTable/>);
 expect(wrapper).toMatchSnapshot();
});