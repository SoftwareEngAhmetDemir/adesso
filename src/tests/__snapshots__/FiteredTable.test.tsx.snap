// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test FilteredTable component 1`] = `
<div>
  <div
    className="tab-content"
    id="pills-tabContent"
  >
    <div
      aria-labelledby="pills-home-tab"
      className="tab-pane fade show active "
      id="pills-home"
      role="tabpanel"
      tabIndex={0}
    >
      <UnitsTables />
    </div>
    <div
      aria-labelledby="pills-profile-tab"
      className="tab-pane fade  "
      id="pills-profile"
      role="tabpanel"
      tabIndex={0}
    >
      <UnitsTables />
    </div>
    <div
      aria-labelledby="pills-contact-tab"
      className="tab-pane fade  "
      id="pills-contact"
      role="tabpanel"
      tabIndex={0}
    >
      <UnitsTables />
    </div>
    <div
      aria-labelledby="pills-disabled-tab"
      className="tab-pane fade  "
      id="pills-disabled"
      role="tabpanel"
      tabIndex={0}
    >
      <UnitsTables />
    </div>
    <div
      aria-labelledby="pills-disabled-tab2"
      className="tab-pane fade  "
      id="pills-disabled2"
      role="tabpanel"
      tabIndex={0}
    >
      <UnitsTables />
    </div>
  </div>
</div>
`;
